js/data/bucket.js

```diff
@@ -344,16 +399,34 @@ Bucket.prototype.recalculateStyleLayers = function() {
     }
 };
 
-Bucket.prototype.populatePaintArrays = function(interfaceName, globalProperties, featureProperties, startGroup, startIndex) {
+
+/**
+ * @typedef {object} ArrayRange
+ * @property {number} arrayRange.startGroup
+ * @property {number} arrayRange.startVertex
+ * @property {number} arrayRange.endGroup
+ * @property {number} arrayRange.endVertex
+ * @private
+ */
+
+Bucket.prototype.populatePaintArrays = function(interfaceName, globalProperties, featureProperties, arrayRange, featureIndex) {
+    if (typeof featureIndex !== 'undefined') {
+        this._featureIndexToArrayRange[featureIndex] = arrayRange;
```

Comment:
It occurs to me that this map should actually be `this._featureIndexToArrayRange[interfaceName][featureIndex]`